Resources/Storage/FileStorage.py

```diff
@@ -0,0 +1,629 @@
+      except OSError as ose:
+        failed[src_url] = str( ose )
+
+    return S_OK( { 'Failed' : failed, 'Successful' : successful } )
+
+
+
+
+
+  def putFile( self, path, sourceSize = 0 ):
+    """Put a copy of the local file to the current directory on the
+       physical storage
+       :param path: dictionnary {pfn  : localFile}
+       :param sourceSize : compares the size of the local and remote.
+                           You obviously run into trouble if you use a list of path...
+                           If the size do not match, remove the remote file
+       :returns Successful dict {path : size}
+                Failed dict {path : error message }
+    """
+
+    if not isinstance( path, dict ):
+      return S_ERROR ( "FileStorage.putFile: path argument must be a dictionary (or a list of dictionary) { url : local path}" )
+
+    failed = {}
+    successful = {}
+
+    for dest_url, src_file in path.items():
+      try:
+        dirname = os.path.dirname( dest_url )
+        if not os.path.exists(dirname):
+          os.makedirs( dirname )
+        shutil.copy2( src_file, dest_url )
+        fileSize = os.path.getsize( dest_url )
+        if sourceSize and ( sourceSize != fileSize ):
+          try:
+            os.unlink(dest_url)
+          except OSError as _ose:
+            pass
+          failed[dest_url] = "Source and destination file sizes do not match (%s vs %s)." % ( sourceSize, fileSize )
+        else:
+          successful[dest_url] = fileSize
+      except OSError as ose:
+        failed[dest_url] = str( ose )
+
+    return S_OK( { 'Failed' : failed, 'Successful' : successful } )
+
+
+
+  def removeFile( self, path ):
+    """Remove physically the file specified by its path
+
+      A non existing file will be considered as successfully removed.
+
+      :param path: path (or list of path) on storage
+      :returns Successful dict {path : True}
+               Failed dict {path : error message }
+    """
+
+    res = checkArgumentFormat( path )
+    if not res['OK']:
+      return res
+    urls = res['Value']
+    gLogger.debug( "FileStorage.removeFile: Attempting to remove %s files." % len( urls ) )
+
+    failed = {}
+    successful = {}
+
+    for url in urls:
+      try:
+        os.unlink(url)
+        successful[url] = True
+      except OSError as ose:
+        # Removing a non existing file is a success
+        if ose.errno == errno.ENOENT:
+          successful[url] = True
+        else:
+          failed[url] = str( ose )
+      except Exception as e:
+        failed[url] = str( e )
+
+    return S_OK( { 'Failed' : failed, 'Successful' : successful } )
+
+
+
+  @staticmethod
+  def __stat( path ):
+    """  Issue a stat call and format it the dirac way, and add the checksum
+
+      :param self: self reference
+      :param path: path on the storage
+      :returns Successful S_OK(metadataDict) or S_ERROR
+    """
+    try:
+      statInfo = os.stat( path )
+      metadataDict = {}
```

Comment:
Probably it would be worth returning all information usually returned by SRM (Unavailable, Cached etc...) as I am not sure we are protected everywhere with .get()